DAQ_System/Source/LSM303DLHC.cpp

```diff
@@ -0,0 +1,136 @@
+#include "mbed.h"
+#include "LSM303DLHC.h"
+#include "stdio.h"
+#include <locale>
+ 
+#define MAG_ADDRESS  0x3C
+#define ACC_ADDRESS  0x32
+  
+ 
+LSM303DLHC::LSM303DLHC(PinName sda, PinName scl): _device(sda, scl){
+    _device.frequency(400000);
+}
+ 
+void LSM303DLHC::init()
+{
+    // init mag
+    // continuous conversion mode
+    _data[0] = MR_REG_M;
+    _data[1] = 0x00;
+    _device.write(MAG_ADDRESS, _data, 2);
+    // data rate 75hz
+    _data[0] = CRA_REG_M;
+    _data[1] = 0x18; // 0b00011000
+    _device.write(MAG_ADDRESS, _data, 2);
+    // init acc
+    // data rate 100hz 
+    _data[0] = CTRL_REG1_A;
+    _data[1] = 0x2F; // 0b00101111
+    _device.write(ACC_ADDRESS, _data, 2);
+}
+ 
+void LSM303DLHC::read(int a[3], int m[3])
+{
+    readAcc(a);
+    readMag(m);   
+}
+ 
+void LSM303DLHC::readAcc(int a[3])
```

Comment:
Hm...thoughts on renaming this to `ReadNominalAcceleration()` or `ReadRawAcceleration()`?